framework/src/base/Moose.C

```diff
@@ -478,6 +482,7 @@ associateSyntaxInner(Syntax & syntax, ActionFactory & /*action_factory*/)
   registerSyntax("AddFieldSplitAction", "Preconditioning/*/*");
 
   registerSyntax("CreateExecutionerAction", "Executioner");
+  registerSyntaxTask("AddSolveObjectAction", "Executioner/*", "add_solve_object");
```

Comment:
The easiest way to keep the solve objects separate from the other objects inside the executioner is to change this to `Executioner/SolveObjects/*`. It will also remove the need for that if statement in AddSolveObjectAction. Also, with the `Executioner/*` syntax if someone misspells Adaptivity (or TimeStepper, etc.), they will get a weird error message saying the object is not a registered SolveObject.

Reply:
In a long run, we want SolveObject to be added by `Executioner/*` syntax. I tried `Executioner/SolveObjects/*` and gave up that. I prefer to add better error messages in such cases instead of compromise our design.

Reply:
Could you explain a bit more why "we" want this syntax? I apologize if I missed a conversation at some point.

Reply:
This is almost by intuition and we did not discuss this in detail. Executioners add solve objects thus solve objects is under one level down in the input graph. There is no reason to have an intermediate level `SolveObjects` unless we are forced to due to the current interventions of other syntax.

Reply:
Let me look more into the MooseDocs code to see what would be required to get the documentation to show up properly. Personally, I find it intuitive that defining solvers would be under a `SolveObjects/` or `Solvers/` branch.

Reply:
`Executioner` is solver. I have pretty strong opinion on not have another layer in the input, it is affecting the look of our depletion executioner.